### PR TITLE
[4.6] docs(charts): add charts trendline documentation

### PR DESCRIPTION
**Pull request by xristianstefanov**
Related to: https://github.com/telerik/blazor/issues/6388.

### CODE REVIEW

File: components/chart/trendlines.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+| `ChartSeriesTrendlineForecast.Before` | `int` | The number of intervals to extend the trendline before the first data point. |
+| `ChartSeriesTrendlineForecast.After` | `int` | The number of intervals to extend the trendline after the last data point. |
+| `ChartSeriesTrendline.Period` | `int` | The number of intervals to take when calculating averages. The value should be an integer greater than 2. |
+| `Type` | `ChartSeriesType (enum) ` | The type of the series. |
+
+## Supported Series Types
+
+Trendlines are supported for the following chart types:
+
+| Chart Type |
```
Comment (dimodi) [inline components/chart/trendlines.md:304]: Why table? A list is more efficient and accessible markup.

File: components/chart/trendlines.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+`````
+
+## Chart Trendlines Parameters
+
+The following table lists Chart Trendlines parameters.
+
+| Parameter | Type and Default&nbsp;Value | Description |
+| --- | --- | --- |
+| `ChartSeries.For` | `string` | The name of the parent series of the trendline. |
+| `ChartSeriesTrendline` | `Object` | The trendline configuration options. |
```
Comment (dimodi) [inline components/chart/trendlines.md:293]:
```suggestion
| `ChartSeriesTrendline` | `object` | The trendline configuration options. |
```

File: components/chart/trendlines.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+| `ChartSeriesTrendlineForecast` | `Object` | The trendline forecast settings. By default, the trendline does not display a forecast. |
+| `ChartSeriesTrendlineForecast.Before` | `int` | The number of intervals to extend the trendline before the first data point. |
+| `ChartSeriesTrendlineForecast.After` | `int` | The number of intervals to extend the trendline after the last data point. |
+| `ChartSeriesTrendline.Period` | `int` | The number of intervals to take when calculating averages. The value should be an integer greater than 2. |
+| `Type` | `ChartSeriesType (enum) ` | The type of the series. |
```
Comment (dimodi) [inline components/chart/trendlines.md:298]:
```suggestion
| `Type` | `ChartSeriesType` enum | The type of the series. |
```

File: components/chart/trendlines.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        }
+    }
+}
+`````
+
+## Chart Trendlines Parameters
+
+The following table lists Chart Trendlines parameters.
+
+| Parameter | Type and Default&nbsp;Value | Description |
```
Comment (dimodi) [inline components/chart/trendlines.md:290]: If all default values match the default value of the type, then remove "and Default Value" as you don't have any.

File: components/chart/trendlines.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+        }
+    }
+}
+`````
+
+### Moving Average Trendline
+
+The Moving Average Trendline smoothens out data fluctuations by computing an average of all data points over a specified period. By default, this period is set to `2` chart intervals.
+
+>caption The following example showcases a moving average trendline in line series:
```
Comment (dimodi) [inline components/chart/trendlines.md:163]: Captions don't need to be like sentences.

```suggestion
>caption Line Chart with a moving average trendline
```

File: components/chart/trendlines.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+    }
+
+    public class SalesData
+    {
+        public string Period { get; set; }
+        public int NumericPeriod { get; set; }
+        public DateTime Date { get; set; }
+        public int Count { get; set; }
+
+        public static List<SalesData> GenerateData()
```
Comment (dimodi) [inline components/chart/trendlines.md:225]: It's strange architecture to hard-code data in a the model class. Move this to a separate method and call it from `OnInitializedAsync`.

File: components/chart/trendlines.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+        </ChartCategoryAxis>
+    </ChartCategoryAxes>
+
+    <ChartValueAxes>
+        <ChartValueAxis Min="-4e5" Max="1.2e6"></ChartValueAxis>
+    </ChartValueAxes>
+</TelerikChart>
+
+@code {
+    public const string Height = "400px";
```
Comment (dimodi) [inline components/chart/trendlines.md:206]: Use `private`, according to code style guide.

Consider hard-coding the Width and Height in the Chart if you use them just once and don't change them at runtime.

File: components/chart/trendlines.md [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+tags: telerik,blazor,chart,trendlines
+published: true
+position: 25
+---
+
+# Trendlines
+
+Trendlines are dynamic indicators that automatically reveal the overarching trends within your series data. These trendlines are defined as a special type of series linked to the main series by name.
+
+Trendline series use the data from the main series. In cases where the main series employs aggregates, which is common for date series, the trendlines align with the aggregated data. For instance, when using a `sum` aggregate, the trendline illustrates the trends of the summations within each category.
```
Comment (dimodi) [inline components/chart/trendlines.md:15]:
```suggestion
Trendline series use the data from the main series. When the main series employs aggregates, which is common for date series, the trendlines align with the aggregated data. For instance, when using a `sum` aggregate, the trendline illustrates the trends for the sum within each category.
```

File: components/chart/trendlines.md [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+
+## Trendline Types
+
+The Chart supports two fundamental types of trendlines - the `Linear Trendline` and the `Moving Average Trendline`.
+
+### Linear Trendline
+
+The Linear Trendline serves the purpose of indicating whether a specific quantity is on the rise or decline over time.
+
+>caption Below is an illustrative example showcasing the creation of a linear trendline in scatter line series:
```
Comment (dimodi) [inline components/chart/trendlines.md:25]:
```suggestion
>caption Linear trendline in scatter line Chart series
```

**Review by dimodi — CHANGES_REQUESTED** [review]
* Captions should look like headings rather than sentences.
* Follow docs code style and refactor the data generation.
* Consider a list instead of a one-column table.
* Polish trendline parameter table.

File: components/chart/trendlines.md [hunk 10]
```diff
@@ -0,0 +1,320 @@
+---
+title: Trendlines
+page_title: Chart Trendlines
+description: Trendlines in the Charts for Blazor.
```
Comment (yordan-mitev) [inline components/chart/trendlines.md:4]: The [meta description](https://docs.telerik.com/style-guide/metadata#guidelines) must be at least 100 symbols long. The optimal length is 150. Try building one or two sentences that serve as a Call-to-Action (CTA).

File: components/chart/trendlines.md [hunk 11]
```diff
@@ -0,0 +1,320 @@
+---
+title: Trendlines
+page_title: Chart Trendlines
```
Comment (yordan-mitev) [inline components/chart/trendlines.md:3]: The rest of the articles in this section use the pattern "Chart" - [article topic].
```suggestion
page_title: Chart - Trendlines
```

File: components/chart/trendlines.md [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+The following table lists Chart Trendlines parameters.
+
+| Parameter | Type and Default&nbsp;Value | Description |
+| --- | --- | --- |
+| `ChartSeries.For` | `string` | The name of the parent series of the trendline. |
+| `ChartSeriesTrendline` | `Object` | The trendline configuration options. |
+| `ChartSeriesTrendlineForecast` | `Object` | The trendline forecast settings. By default, the trendline does not display a forecast. |
+| `ChartSeriesTrendlineForecast.Before` | `int` | The number of intervals to extend the trendline before the first data point. |
+| `ChartSeriesTrendlineForecast.After` | `int` | The number of intervals to extend the trendline after the last data point. |
+| `ChartSeriesTrendline.Period` | `int` | The number of intervals to take when calculating averages. The value should be an integer greater than 2. |
```
Comment (yordan-mitev) [inline components/chart/trendlines.md:297]:
```suggestion
| `ChartSeriesTrendline.Period` | `int` | The number of intervals to take when calculating averages. The value must be an integer greater than `2`. |
```

**Pull request merged**